byterun/sys.c

```diff
@@ -336,11 +336,13 @@ CAMLprim value caml_sys_chdir(value dirname)
 CAMLprim value caml_sys_getcwd(value unit)
 {
   char buff[4096];
```

Comment:
This is perhaps slightly off-topic, but shouldn't the length of `buff` use `PATH_MAX` rather than `4096`?

Reply:
Well, for modern OSes and file systems, there is no fixed limit on the length of an absolute file path, so 4096 is not much more wrong than `PATH_MAX`...

We probably could do a better job by dynamically resizing the buffer if `getcwd` errors with `ERANGE`.  But that's a topic for another pull request.